client/client.go

```diff
@@ -239,6 +246,138 @@ func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) error
 	return err
 }
 
+func searchSlice(slice []string, key string) bool {
+	for _, v := range slice {
+		if v == key {
+			return true
+		}
+	}
+	return false
+}
+
+// DoWithFilters performs request passed and filters entities in json response
+func (c *Client) DoWithFilters(ctx context.Context, req *http.Request, v interface{}, filters []*AdditionalFilter, baseSearchPaths [][]string) error {
+	req = req.WithContext(ctx)
+	resp, err := c.client.Do(req)
+
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if rerr := resp.Body.Close(); err == nil {
+			err = rerr
+		}
+	}()
+
+	err = CheckResponse(resp)
+
+	if err != nil {
+		return err
+	}
+
+	resp.Body = filter(resp.Body, filters, baseSearchPaths)
+
+	if v != nil {
+		if w, ok := v.(io.Writer); ok {
+			_, err = io.Copy(w, resp.Body)
+			if err != nil {
+				fmt.Printf("Error io.Copy %s", err)
+
+				return err
+			}
+		} else {
+			err = json.NewDecoder(resp.Body).Decode(v)
+			if err != nil {
+				return fmt.Errorf("error unmarshalling json: %s", err)
+			}
+		}
+	}
+
+	if c.onRequestCompleted != nil {
+		c.onRequestCompleted(req, resp, v)
+	}
+
+	return err
+}
+
+func filter(body io.ReadCloser, filters []*AdditionalFilter, baseSearchPaths [][]string) io.ReadCloser {
+	if filters == nil {
+		return body
+	}
+	var res map[string]interface{}
+	b, _ := io.ReadAll(body)
+	json.Unmarshal(b, &res)
+
+	// Full search paths
+	searchPaths := map[string][][]string{}
+
+	filterMap := map[string]*AdditionalFilter{}
+	for _, filter := range filters {
+		filterMap[filter.Name] = filter
+
+		// Build search paths by appending target search paths to base paths
+		filterSearchPaths := [][]string{}
+		if len(baseSearchPaths) == 0 {
+			searchPath := strings.Split(filter.Name, ".")
+			filterSearchPaths = append(filterSearchPaths, searchPath)
+		}
+		for _, baseSearchPath := range baseSearchPaths {
+			searchPath := append(baseSearchPath, strings.Split(filter.Name, ".")...)
+			filterSearchPaths = append(filterSearchPaths, searchPath)
+		}
+		searchPaths[filter.Name] = filterSearchPaths
+	}
+
+	// Entities that pass filters
+	var filteredEntities []interface{}
+
+	entities := res["entities"].([]interface{})
+	for _, entity := range entities {
```

Comment:
Can we bring down the complexity of this loop ? 
We can also think of using external library like [jsonpath](https://github.com/PaesslerAG/jsonpath)